src/chaat/app.clj

```diff
@@ -1,30 +1,73 @@
 (ns chaat.app
   (:require [ring.middleware.reload :refer [wrap-reload]]
+            [ring.middleware.keyword-params :refer [wrap-keyword-params]]
+            [ring.middleware.params :refer [wrap-params]]
+            [ring.middleware.json :refer [wrap-json-params]]
             [ring.adapter.jetty :as jetty]
             [chaat.routes :as routes]
-            [bidi.ring :refer [make-handler]])
+            [bidi.ring :refer [make-handler]]
+            [chaat.migrations :as migrations]
+            [chaat.config :as config]
+            [next.jdbc.connection :as conn]
+            [com.stuartsierra.component :as component]
+            [ragtime.core])
+  (:import (com.zaxxer.hikari HikariDataSource))
   (:gen-class))
 
-(def handler
-  (make-handler routes/routes))
-
-(def app
-  (-> #'handler
+;; handler + middleware
+(defn make-app [db]
+  (-> (routes/routes-fn db)
+      make-handler
+      wrap-keyword-params
+      wrap-json-params
+      wrap-params
       wrap-reload))
 
-(defonce server (atom nil))
+;; defining HttpServer component
+(defrecord HttpServer [config db]
+  component/Lifecycle
+  (start [component]
+    (if (:server component)
+      component
+      (let [options (:options config)
+            server (jetty/run-jetty (make-app db) options)]
+        (assoc component :server server))))
+  (stop [component]
+    (if-let [server (:server component)]
+      (do (.stop server)
+          (.join server)
+          (dissoc component :server))
+      component)))
 
-(defn start-server
-  [val]
-  (jetty/run-jetty app {:port 3000
-                        :join? false}))
+;; constructor for HttpServer
+(defn new-http-server [config]
```

Comment:
suggestion: this function seems redundant. lets try to keep this within the new-system call

Reply:
Agreed about the redundancy. I would like to leave this in until we next speak, have a few questions about this.